lib/si-layer-cache/src/layer_cache.rs

```diff
@@ -18,66 +21,61 @@ pub struct LayerCache<V>
 where
     V: Serialize + DeserializeOwned + Clone + Send + Sync + 'static,
 {
-    memory_cache: MemoryCache<V>,
-    disk_cache: DiskCache,
+    cache: Cache<V>,
+    name: String,
     pg: PgLayer,
-    #[allow(dead_code)] // TODO(fnichol): remove once in use
     compute_executor: DedicatedExecutor,
 }
 
 impl<V> LayerCache<V>
 where
     V: Serialize + DeserializeOwned + Clone + Send + Sync + 'static,
 {
-    pub fn new(
+    pub async fn new(
         name: &str,
         pg_pool: PgPool,
-        memory_cache_config: MemoryCacheConfig,
-        disk_cache_config: DiskCacheConfig,
+        cache_config: CacheConfig,
         compute_executor: DedicatedExecutor,
-    ) -> LayerDbResult<Self> {
-        let disk_cache = DiskCache::new(disk_cache_config)?;
+        tracker: TaskTracker,
+        token: CancellationToken,
+    ) -> LayerDbResult<Arc<Self>> {
+        let cache = Cache::new(cache_config).await?;
 
         let pg = PgLayer::new(pg_pool.clone(), name);
 
-        Ok(LayerCache {
-            memory_cache: MemoryCache::new(memory_cache_config),
-            disk_cache,
+        let lc: Arc<LayerCache<V>> = LayerCache {
+            cache,
+            name: name.to_string(),
             pg,
             compute_executor,
-        })
+        }
+        .into();
+
+        tracker.spawn(lc.clone().shutdown_handler(token.clone()));
+        Ok(lc)
     }
 
-    async fn spawn_disk_cache_write_vec(&self, key: Arc<str>, value: Vec<u8>) -> LayerDbResult<()> {
-        self.disk_cache().insert(key, value).await?;
+    async fn shutdown_handler(self: Arc<Self>, token: CancellationToken) -> LayerDbResult<()> {
+        token.cancelled().await;
+        debug!("shutting down layer cache {}", self.name);
+        self.compute_executor.shutdown();
```

Comment:
You should be able to skip shutting down the compute executor. I had to double check the impl, but calling `shutdown` will trigger the shutdown of all clones of the executor so there could be a shutdown race between this code and other other code that is using the same cloned executor. I was looking for where in layer-cache we were shutting this down, then didn't find it which made me check.

Reply:
I'll remove it. Should shutdown calls be dedup'd and no-op if it's called more than once? Otherwise you kind of need to know that this has been cloned and shutdown elsewhere